core/src/apps/management/recovery_device/layout.py

```diff
@@ -48,12 +48,38 @@ async def request_mnemonic(
 
     await button_request("mnemonic", code=ButtonRequestType.MnemonicInput)
 
+    # Allowing to go back to previous words, therefore cannot use just loop over range(word_count)
     words: list[str] = []
```

Comment:
`words = [""] * word_count` simplifies a lot of the special cases below

Reply:
Right, thanks for the idea. Done in [33ce5da](https://github.com/trezor/trezor-firmware/pull/3458/commits/33ce5da629fc0aeb862c95a9ebe04027ef058fa2)